File: zed.py


print("Starting ZED camera")

import pyzed.sl as sl
# Create a Camera object
# zed = sl.Camera()
# exit()
# # Set initialization parameters
# init_params = sl.InitParameters()
# init_params.camera_resolution = sl.RESOLUTION.HD720
# init_params.camera_fps = 30

# # Open the camera
# if zed.open(init_params) != sl.ERROR_CODE.SUCCESS:
#     print("Error opening camera")
#     exit()

# # Capture images in a loop
# image = sl.Mat()
# while True:
#     if zed.grab() == sl.ERROR_CODE.SUCCESS:
#         # Retrieve the left image
#         zed.retrieve_image(image, sl.VIEW.LEFT)
#         # Process and display image
#         # ...

# # Close the camera
# zed.close()

File: zed.cpp
#include <sl/Camera.hpp>

int main(int argc, char **argv) {
    // Create a ZED camera object

    std::cout << "HELLO WORLD" << std::endl;

    sl::Camera zed;

    // Set configuration parameters
    sl::InitParameters init_params;
    init_params.camera_resolution = sl::RESOLUTION::HD720; // Set the resolution
    init_params.camera_fps = 30; // Set the FPS

    // Open the camera
    sl::ERROR_CODE err = zed.open(init_params);
    if (err != sl::ERROR_CODE::SUCCESS) {
        std::cout << "Failed to open camera" << std::endl;
        return 1;
    }

    // Capture images in a loop
    sl::Mat image;
    while (true) {
        // Grab a new image
        if (zed.grab() == sl::ERROR_CODE::SUCCESS) {
            // Retrieve the left image
            zed.retrieveImage(image, sl::VIEW::LEFT);
            // Display the image using OpenCV or other display methods
            // ...
        }
    }

    // Close the camera
    zed.close();
    return 0;
}
